src/main/java/inventorysetups/InventorySetupsPersistentDataManager.java

```diff
@@ -148,6 +163,53 @@ private <T> List<T> loadData(final String configKey, Type type)
 		}
 	}
 
+	private List<InventorySetup> loadV1Setups() {
+		Type setupTypeV1 = new TypeToken<ArrayList<InventorySetup>>()
+		{
+
+		}.getType();
+		return loadData(CONFIG_KEY_SETUPS, setupTypeV1);
+	}
+
+	private List<InventorySetup> loadV2Setups() {
+		Type setupTypeV2 = new TypeToken<ArrayList<InventorySetupSerializable>>()
+		{
+
+		}.getType();
+		List<InventorySetupSerializable> issList = new ArrayList<>(loadData(CONFIG_KEY_SETUPS_V2, setupTypeV2));
+		List<InventorySetup> isList = new ArrayList<>();
+		for (final InventorySetupSerializable iss : issList)
+		{
+			isList.add(InventorySetupSerializable.convertToInventorySetup(iss));
+		}
+		return isList;
+	}
+
+	private List<InventorySetup> loadV3Setups() {
+		final String wholePrefix = ConfigManager.getWholeKey(CONFIG_GROUP, null, CONFIG_KEY_SETUPS_V3_PREFIX);
+		final List<String> loadedSetupWholeKeys = configManager.getConfigurationKeys(wholePrefix);
+		Set<String> loadedSetupKeys = loadedSetupWholeKeys.stream().map(
+			key -> key.substring(wholePrefix.length()-CONFIG_KEY_SETUPS_V3_PREFIX.length())
+		).collect(Collectors.toSet());
+
+		List<InventorySetup> loadedSetups = new ArrayList<>();
+		for (final String configKey : loadedSetupKeys)
+		{
+			final String storedData = configManager.getConfiguration(CONFIG_GROUP, configKey);
+
+			try
+			{
+				loadedSetups.add(InventorySetupSerializable.convertToInventorySetup(gson.fromJson(storedData, InventorySetupSerializable.class)));
+			}
+			catch (Exception e)
+			{
+				log.error(String.format("Exception occurred while loading %s", configKey), e);
+				throw e;
+			}
+		}
+		return loadedSetups;
+	}
+

```

Comment:
Line 173/235 is now irrelevant. If someone has previous duplicate name setups, they will get overwritten in `updateConfig` and this code won't do anything. We could fix this by checking for duplicate names during V1/V2 migration, but This is probably not necessary since it's been a very long time since I fixed the duplicate names.

```
final String newName = InventorySetupUtilities.findNewName(setup.getName(), cache.getInventorySetupNames().keySet());
setup.setName(newName);
```

You can delete these lines. I tested it by creating a duplicate entry in the config (with duplicate name), and all that happens is one of the duplicates gets overwritten.

Reply:
done